testnet/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -585,10 +585,41 @@ impl UTXO {
     }
 
     pub fn get_sat_amount(&self) -> u64 {
-        let res = self.amount * 10e7;
-        res as u64
+        UTXO::serialized_btc_to_sat(&self.amount)
     }
 
+    pub fn serialized_btc_to_sat(amount: &str) -> u64 {
+        let comps: Vec<&str> = amount.split(".").collect();
+        match comps[..] {
+            [lhs, rhs] => {
+                let base: u64 = 10;
+                let sat_decimals = 7;
+                let btc_to_sat = base.pow(7);
+                let btc = lhs.parse::<u64>().expect("Invalid amount");
+                let mut amount = btc * btc_to_sat;
+                if rhs.len() > sat_decimals { 
+                    panic!("Unexpected amount of decimals");
+                }
+                let rhs = format!("{:0<width$}", rhs, width = sat_decimals);
+                let sat = rhs.parse::<u64>().expect("Invalid amount");
+                amount += sat;
+                amount
+            },
+            _ => panic!("Invalid amount")
+        }    
+    } 
+
+    pub fn sat_to_serialized_btc(amount: u64) -> String {
```

Comment:
Why not just do this?
```
let int_part = amount / 10.pow(8);
let frac_part = amount % 10.pow(8);
format!("{}.{}", int_part, frac_part)
```

Reply:
for `100000001`, I think your snippet would print `1.1` - you need to pad

Reply:
Yup; updated the example